internal/cmd/grants.go

```diff
@@ -245,6 +250,54 @@ func addGrant(cli *CLI, cmdOptions grantsCmdOptions) error {
 	return nil
 }
 
+func checkResourcesPrivileges(client *api.Client, resource, privilege string) error {
+	parts := strings.SplitN(resource, ".", 2)
+	destination := parts[0]
+	subresource := ""
+
+	if len(parts) > 1 {
+		subresource = parts[1]
+	}
+
+	supportedResources := make(map[string]struct{})
+	supportedRoles := make(map[string]struct{})
+
+	if destination != "infra" {
+		destinations, err := client.ListDestinations(api.ListDestinationsRequest{Name: destination})
+		if err != nil {
+			return err
+		}
+
+		if destinations.Count == 0 {
+			return fmt.Errorf("unknown destination %q", destination)
+		}
+
+		for _, d := range destinations.Items {
+			for _, r := range d.Resources {
+				supportedResources[r] = struct{}{}
```

Comment:
If it's just one resource and one role, we don't need to build a map, you can just use a single `bool` variable for each

Reply:
The map is for multiple destinations that may match the name query. Since the returned value is a list of destinations, it's better to build and use this set